mp/src/game/client/hud_damageindicator.cpp

```diff
@@ -0,0 +1,332 @@
+        int iHeight = RemapVal(clampedDamage, 0, m_iMaximumDamage, m_flMinimumHeight, m_flMaximumHeight) * 0.5;
+
+        // Find the place to draw it
+        float xpos, ypos;
+        float flRotation;
+        float flTimeSinceStart = (gpGlobals->curtime - m_vecDamages[i].flStartTime);
+        GetDamagePosition(m_vecDamages[i].vecDelta, m_flRadius, &xpos, &ypos, &flRotation);
+
+        // Calculate life left
+        float flLifeLeft = (m_vecDamages[i].flLifeTime - gpGlobals->curtime);
+        if (flLifeLeft > 0)
+        {
+            float flPercent = flTimeSinceStart / (m_vecDamages[i].flLifeTime - m_vecDamages[i].flStartTime);
+            float alpha;
+            if (flPercent <= m_flFadeOutPercentage)
+            {
+                alpha = 1.0;
+            }
+            else
+            {
+                alpha = 1.0 - RemapVal(flPercent, m_flFadeOutPercentage, 1.0, 0.0, 1.0);
+            }
+            DrawDamageIndicator(xpos - iWidth, ypos - iHeight, xpos + iWidth, ypos + iHeight, alpha, flRotation);
+        }
+        else
+        {
+            m_vecDamages.Remove(i);
+        }
+    }
+}
+
+//-----------------------------------------------------------------------------
+// Purpose: Message handler for Damage message
+//-----------------------------------------------------------------------------
+void CHudDamageIndicator::MsgFunc_DamageIndicator(bf_read &msg)
+{
+    // Update the convars
+    m_flMinimumWidth = maxwidth.GetFloat();
+    m_flMaximumWidth = maxwidth.GetFloat();
+    m_flMinimumHeight = minheight.GetFloat();
+    m_flMaximumHeight = maxheight.GetFloat();
+    m_flRadius = radius.GetFloat();
+    m_flMinimumTime = lifetime.GetFloat();
+    m_flMaximumTime = lifetime.GetFloat();
+
+    damage_t damage;
+    damage.iScale = msg.ReadByte();
+    if (damage.iScale > m_iMaximumDamage)
+    {
+        damage.iScale = m_iMaximumDamage;
+    }
+    Vector vecOrigin;
+    msg.ReadBitVec3Coord(vecOrigin);
+    damage.flStartTime = gpGlobals->curtime;
+    damage.flLifeTime =
+        gpGlobals->curtime + RemapVal(damage.iScale, 0, m_iMaximumDamage, m_flMinimumTime, m_flMaximumTime);
+    if (vecOrigin == vec3_origin)
+    {
+        vecOrigin = MainViewOrigin();
+    }
+
+    damage.vecDelta = (vecOrigin - MainViewOrigin());
+    VectorNormalize(damage.vecDelta);
+
+    m_vecDamages.AddToTail(damage);
+}
+
+//-----------------------------------------------------------------------------
+// Purpose: hud scheme settings
+//-----------------------------------------------------------------------------
+void CHudDamageIndicator::ApplySchemeSettings(vgui::IScheme *pScheme)
+{
+    BaseClass::ApplySchemeSettings(pScheme);
+    SetPaintBackgroundEnabled(false);
+
+    // set our size
+    int screenWide, screenTall;
+    int x, y;
+    GetPos(x, y);
+    GetHudSize(screenWide, screenTall);
+    SetBounds(0, y, screenWide, screenTall - y);
```

Comment:
Setting the size should happen in `PerformLayout()` so override that and move this sizing code there.